icu4j/main/tests/core/src/com/ibm/icu/dev/test/number/NumberRangeFormatterTest.java

```diff
@@ -966,6 +966,54 @@ public void test21358_SignPosition() {
         }
     }
 
+    @Test
+    public void testCreateLNRFFromNumberingSystemInSkeleton() {
+        {
+            LocalizedNumberRangeFormatter lnrf = NumberRangeFormatter
+                .withLocale(ULocale.forLanguageTag("en"))
+                .numberFormatterBoth(NumberFormatter.forSkeleton(
+                    ".### rounding-mode-half-up"));
+            String actual = lnrf.formatRange(1, 234).toString();
+            assertEquals("default numbering system", "1–234", actual);
+        }
+        {
+            LocalizedNumberRangeFormatter lnrf = NumberRangeFormatter
+                .withLocale(ULocale.forLanguageTag("th"))
+                .numberFormatterBoth(NumberFormatter.forSkeleton(
+                    ".### rounding-mode-half-up numbering-system/thai"));
+            String actual = lnrf.formatRange(1, 234).toString();
+            assertEquals("Thai numbering system", "๑-๒๓๔", actual);
+        }
+        {
+            LocalizedNumberRangeFormatter lnrf = NumberRangeFormatter
+                .withLocale(ULocale.forLanguageTag("en"))
+                .numberFormatterBoth(NumberFormatter.forSkeleton(
+                    ".### rounding-mode-half-up numbering-system/arab"));
+            String actual = lnrf.formatRange(1, 234).toString();
+            assertEquals("Arabic numbering system", "١–٢٣٤", actual);
+        }
+        {
+            LocalizedNumberRangeFormatter lnrf = NumberRangeFormatter
+                .withLocale(ULocale.forLanguageTag("en"))
+                .numberFormatterFirst(NumberFormatter.forSkeleton("numbering-system/arab"))
+                .numberFormatterSecond(NumberFormatter.forSkeleton("numbering-system/arab"));
+            String actual = lnrf.formatRange(1, 234).toString();
+            assertEquals("Double Arabic numbering system", "١–٢٣٤", actual);
+        }
+        {
+            LocalizedNumberRangeFormatter lnrf = NumberRangeFormatter
+                .withLocale(ULocale.forLanguageTag("en"))
+                .numberFormatterFirst(NumberFormatter.forSkeleton("numbering-system/arab"))
+                .numberFormatterSecond(NumberFormatter.forSkeleton("numbering-system/latn"));
+            try {
```

Comment:
can we separate this block to a different test func and use "@Test(expected = IllegalArgumentException.class)" as suggested in  https://www.baeldung.com/junit-assert-exception#junit-4 to assert the expected throw?